drivers/clock_control/clock_stm32_ll_u3.c

```diff
@@ -0,0 +1,684 @@
+		if (LL_PWR_IsEnabledBkUpAccess() == 0U) {
+			/* Enable write access to Backup domain */
+			LL_PWR_EnableBkUpAccess();
+			while (LL_PWR_IsEnabledBkUpAccess() == 0U) {
+				/* Wait for Backup domain access */
+			}
+		}
+
+		/* Configure driving capability */
+		LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << RCC_BDCR_LSEDRV_Pos);
+
+		if (IS_ENABLED(STM32_LSE_BYPASS)) {
+			/* Configure LSE bypass */
+			LL_RCC_LSE_EnableBypass();
+		}
+
+		/* Enable LSE Oscillator */
+		LL_RCC_LSE_Enable();
+		/* Wait for LSE ready */
+		while (LL_RCC_LSE_IsReady() == 0U) {
+		}
+
+		/* Enable LSESYS additionally */
+		LL_RCC_LSE_EnablePropagation();
+		/* Enforce BackUp domain access is disabled after clock initialization */
+		while (LL_RCC_LSE_IsPropagationReady() == 0U) {
+		}
+		LL_PWR_DisableBkUpAccess();
+	}
+
+	if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+		/* Set MSIS Range */
+		LL_FLASH_SetLatency(LL_FLASH_LATENCY_2);
+		while (LL_FLASH_GetLatency() != LL_FLASH_LATENCY_2) {
+		}
+
+		if (IN_RANGE(STM32_MSIS_RANGE, 0, 3)) {
+			LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE1);
+		} else {
+			LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE2);
+		}
+		/* Enable Epod Booster */
+		LL_RCC_SetEPODBoosterClkSource(LL_RCC_EPODBOOSTCLKSRCE_MSIS);
+		LL_RCC_SetEPODBoosterClkPrescaler(LL_RCC_EPODBOOSTCLKPRESCAL_1);
+		LL_PWR_EnableEPODBooster();
+		while (LL_PWR_IsActiveFlag_BOOST() == 0) {
+		}
+
+		/* Activate MSIS as source */
+		LL_RCC_MSIS_Enable();
+		if (IN_RANGE(STM32_MSIS_RANGE, 0, 3)) {
+			/* Range 0-3 uses RC0 as the clock source */
+			LL_RCC_MSIS_SetClockSource(LL_RCC_MSIS_CLOCK_SOURCE_RC0);
+			switch (STM32_MSIS_RANGE) {
+			case 0:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_1);
+				break;
+			case 1:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_2);
+				break;
+			case 2:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_4);
+				break;
+			case 3:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_8);
+				break;
+			default:
+				break;
+			}
+		} else {
+			/* Range 4-7 uses RC1 as the clock source */
+			LL_RCC_MSIS_SetClockSource(LL_RCC_MSIS_CLOCK_SOURCE_RC1);
+			switch (STM32_MSIS_RANGE) {
+			case 4:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_1);
+				break;
+			case 5:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_2);
+				break;
+			case 6:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_4);
+				break;
+			case 7:
+				LL_RCC_MSIS_SetClockDivision(LL_RCC_MSIS_CLOCK_SOURCE_RC_DIV_8);
+				break;
+			default:
+				break;
+			}
+
+		}
+		LL_RCC_MSI_SetMSIxClockRange();
+		/*
+		 * For stm32u3 LSE or HSE must be enabled and ready before
+		 * selecting this oscillator as MSIRC0/MSIRC1 input clock.
+		 */
+		/* Enable MSIS */
+		LL_RCC_MSIS_Enable();
+		while (LL_RCC_MSIS_IsReady() == 0U) {
+		}
+		LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_MSIS);
+
```

Comment:
Club this line with related while block
```suggestion

		LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_MSIS);
```